### ISSUE TITLE
Metavariable naming bug when case splitting

### ISSUE THREAD

**Issue by dkasak** [idris-lang/Idris-dev#1878]
Given the following code

    import Data.Vect
    
    foo : Vect n a -> Vect n a -> Vect n a
    foo xs ys = ?foo_rhs

splitting on `xs` results in this

    foo : Vect n a -> Vect n a -> Vect n a
    foo [] ys = ?foo_rhs_1
    foo (x :: xs) ys = ?foo_rhs_2

which is the expected result. However, splitting again on `ys` results in this

    foo : Vect n a -> Vect n a -> Vect n a
    foo [] [] = ?foo_rhs_2
    foo (x :: xs) ys = ?foo_rhs_2

which is clearly wrong since the metavariables shouldn't be the same (and differs from the result in this [blog post](https://edwinb.wordpress.com/2013/10/28/interactive-idris-editing-with-vim/)).

**Comment by DavidHarrison** [comment]
Are you splitting on both ys? When I try this (in Idris 0.9.16 with [idris-vim](https://github.com/idris-hackers/idris-vim)), I get to the third set of cases by splitting on xs and the first set of ys, but get to what would be expected from the blog by then splitting on the second set of ys.

**Comment by dkasak** [comment]
No, I'm splitting on the `xs` and then on the `ys` of the first equation. It still seems to me that this shouldn't result in an inconsistent set of cases and, if I'm reading the blog right, this isn't the behaviour displayed there.

Specifically, the blog splits `x2` (the equivalent of my `ys`) here

    vadd : Num a => Vect n a -> Vect n a -> Vect n a
    vadd [] x2 = ?vadd_rhs_1
    vadd (x :: xs) x2 = ?vadd_rhs_2

which results in

    vadd : Num a => Vect n a -> Vect n a -> Vect n a
    vadd [] [] = ?vadd_rhs_3
    vadd (x :: xs) x2 = ?vadd_rhs_2

Notice that this split resulted in `?vadd_rhs_1` changing into `?vadd_rhs_3` so the metavariable in the two equations.are still separate.

I forgot to mention before, but I have the same setup as yours (Idris 0.9.16 and idris-vim from the repo).

**Comment by DavidHarrison** [comment]
Sorry, I had been paying more attention to the case splitting rather than the metavariable naming, I see the issue now.

**Comment by edwinb** [comment]
This happened because parsing stopped in the middle of the definition so Idris didn't know about the old name. I've recently changed the parser so that it will not stop until the end of the definition, when partially parsing a file, so this appears to work fine now.

This issue had annoyed me too for a while, but never quite enough to do anything about it. I actually fixed it by accident when fixing something else last week :).
